source/config/__init__.py

```diff
@@ -35,14 +39,22 @@
 isAppX=False
 
 #: The active configuration, C{None} if it has not yet been loaded.
-#: @type: ConfigObj
+#: @type: ConfigManager
 conf = None
 
 #: Notifies when the configuration profile is switched.
-#: This allows components to apply changes required by the new configuration.
+#: This allows components and add-ons to apply changes required by the new configuration.
 #: For example, braille switches braille displays if necessary.
 #: Handlers are called with no arguments.
 configProfileSwitched = extensionPoints.Action()
+#: Notifies when NVDA is saving current configuration.
+#: Handlers can listen to "pre" or "post" action to perform tasks prior to or after NVDA's own configuration is saved.
+#: Handlers are called with no arguments.
+configPreSave = extensionPoints.Action()
+configPostSave = extensionPoints.Action()
+#: Notifies when configuration is reloaded from disk or factory defaults are applied.
+#: Handlers are called with a boolean argument indicating whether this is a factory reset (True) or just reloading from disk (False).
+configReset = extensionPoints.Action()

```

Comment:
I think it would be good if this followed the pattern of pre / post. If there is no need to implement pre, please just rename this to `postConfigReset`.